### PR TITLE
Setup online lpython

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
Demo: https://dev.lpython.org/

### CODE REVIEW

File: src/bin/CMakeLists.txt [hunk 1]
```diff
@@ -48,29 +48,24 @@ set_target_properties(lpython PROPERTIES
...
+    # "-s EXPORTED_RUNTIME_METHODS=['cwrap']": Export cwarp. cwarp helps us to call our EMSCRIPTEN_KEEPALIVE functions
+    # "-fsanitize=undefined": Clang's Undefined Behaviour Sanitizer. The LPython parser segfaults.
+        # This option is for debugging, but currently helps avoid the segfault in the parser.
+    # "-s INITIAL_MEMORY=536870912": Start the wasm linear memory with sufficiently large size 512Mb.
 
     # Notes:
     # STANDALONE_WASM is disabling support for exceptions, so it is currently omitted
     # In build_to_wasm.sh, we need CMAKE_CXX_FLAGS_DEBUG="-Wall -Wextra -fexceptions" flags for exception support
-    set(WASM_COMPILE_FLAGS "-g0 -fexceptions")
+    set(WASM_COMPILE_FLAGS "-g0 -fexceptions -fsanitize=undefined")
```
Comment (Shaikh-Ubaid) [inline src/bin/CMakeLists.txt:66]: LPython's parser segfaults in many cases when LPython is build to wasm. It seems LPython uses a different parser as compared to LFortran. I think LPython uses LR(1) parser where as LFortran uses GLR parser.

The above `-fsanitize` flag is actually a debugging flag, but it seems to make LPython's parser to work.

File: src/libasr/ASR.asdl [hunk 2]
```diff
@@ -164,6 +168,7 @@ abi                   -- External     ABI
     | GFortranModule  --   Yes        GFortran
     | BindC           --   Yes        C
     | BindPython      --   Yes        Python
+    | BindJS          --   Yes        Javascript
```
Comment (Shaikh-Ubaid) [inline src/libasr/ASR.asdl:171]: I added a new decorator for calls to Javascript functions. We previously used the (BindC + implementation=interface) for recognizing the functions to be imported. But it seems some functions from the numpy intrinsic module use these exact parameters (BindC + interface) and thus we are unable to distinguish between these numpy intrinsic functions and the functions that call into Javascript.

File: tests/reference/asr-elemental_01-b58df26.stderr [hunk 3]
```diff
@@ -1,5 +0,0 @@
-semantic error: Arguments do not match for any generic procedure, sin
-  --> tests/../integration_tests/elemental_01.py:10:24
-   |
-10 |         assert abs(sin(sin(array[i])) - result[i]) <= eps
-   |                        ^^^^^^^^^^^^^
```
Comment (Shaikh-Ubaid) [inline tests/reference/asr-elemental_01-b58df26.stderr:5]: These few files got removed. I think these were left over from some previous PR.

**Comment by Shaikh-Ubaid** [comment]
After we merge this, we need to update the commit version at https://dev.lpython.org/.

File: src/bin/CMakeLists.txt (hunk 1 above)
Comment (certik) [reply src/bin/CMakeLists.txt:66]: Yes, LPython uses LALR(1), which should be an easier parser.

File: src/libasr/ASR.asdl (hunk 2 above)
Comment (certik) [reply src/libasr/ASR.asdl:171]: I think that's fine. The same should be done in LFortran.

**Review by certik — APPROVED** [review]
I think that this is fine, thanks!

File: src/bin/CMakeLists.txt (hunk 1 above)
Comment (Shaikh-Ubaid) [reply src/bin/CMakeLists.txt:66]: With a previous version of emscripten, the LPython parser would abort with error in the lines of "large memory ~3.5Gbs requested which is more than the available ~2.4Gbs". With the latest version of emscripten, it says "aborted, runtime error". (I updated my installed `emscripten` to be sure the errors are not due to some old version. )

**Pull request merged**